cpp/memilio/compartments/compartmentalmodel.h

```diff
@@ -126,39 +126,38 @@ struct CompartmentalModel {
     }
 
     /**
-     * @brief get_initial_values returns the initial values for the compartmental populations.
-     * This can be used as initial conditions in an ODE solver
-     * @return the initial populatoins
+     * @brief Returns the initial values for the compartmental populations.
+     * This can be used as initial conditions in an ODE solver.
+     * @return The initial populations.
      */
     Vector<FP> get_initial_values() const
     {
         return populations.get_compartments();
     }
 
     /**
-     * @brief Checks whether the model satisfies all constraints and applies the constraints, if not.
+     * @brief Checks whether the model satisfies all constraints. If not, it changes values to suffice their constraints.
      *
-     * Attention: This function should be used with care. It is necessary for some test problems to run through quickly,
-     *            but in a manual execution of an example, check_constraints() may be preferred. 
-     *            Note that the apply_constraints() function can and will not set model parameters and 
-     *            compartments to meaningful values.
+     * Attention: This function should be used with care. It can not and will not set model parameters and 
+     *            compartments to meaningful values. In most cases it is preferable to use check_constraints,
+     *            and correct values manually before proceeding with the simulation.
+     *            The main usage for apply_constraints is in automated tests using random values for initialization.
      *
-     * @return Returns true if one (or more) constraint(s) were corrected, otherwise false. 
+     * @return Returns true if one or more constraints were corrected, false otherwise. 
      */
     bool apply_constraints()
     {
-        bool corrected = populations.apply_constraints();
         if constexpr (has_apply_constraints_member_function<ParameterSet>::value) {
-            corrected = (corrected || parameters.apply_constraints());
+            return ((int)parameters.apply_constraints() | (int)populations.apply_constraints());
```

Comment:
```suggestion
            // use bitwise instead of logical or, so that both apply_constraint functions are called
            return ((int)parameters.apply_constraints() | (int)populations.apply_constraints());
```